### PR TITLE
Fix the import fails if not enough data

### PR DESCRIPTION
**Pull request by dobaokhanh**

### CODE REVIEW

File: src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java [hunk 1]
```diff
@@ -98,22 +99,22 @@ public DataSample getSample(DataSeries dataSeries, int rows) {
...
-
             if (columns.isEmpty()) {
                 // Auto-detect types from CSVRecords
-                Map<String, String> csvMap = parser.iterator().next().toMap(); // Map<Header, Data>
 
                 List<String> headers = csvSeriesConfig.getHeaders();
+                Map<String, String> csvMap = records
+                        .stream()
+                        .findFirst()
+                        .orElseThrow(IllegalStateException::new)
```
Comment (vilppuvuorinen) [inline src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java:116]: Please use `InputException` from `fi.jubic.quanta.exception` with a descriptive error message. Something like "Cannot import an empty CSV file".

**Review by vilppuvuorinen — CHANGES_REQUESTED** [review]

File: src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java (hunk 1 above)
Comment (dobaokhanh) [reply src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java]: Changed to `InputException`

File: src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java [hunk 2]
```diff
@@ -165,6 +179,9 @@ public DataSample getSample(DataSeries dataSeries, int rows) {
                     .setData(sampleRows)
                     .build();
         }
+        catch (InputException e) {
```
Comment (vilppuvuorinen) [inline src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java:182]: `InputException` is a subclass of `ApplicationException`. There's no need to catch it and rethrow. Just make sure the message string is included on the first go.

**Review by vilppuvuorinen — CHANGES_REQUESTED** [review]

File: src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java (hunk 2 above)
Comment (dobaokhanh) [reply src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java:182]: Change to `() -> new InputException("Can not import empty CSV file.")`

**Review by vilppuvuorinen — APPROVED** [review]

**Pull request merged**